src/BrazeKit-dev.js

```diff
@@ -66,32 +66,92 @@ var constructor = function () {
         dob: 'setDateOfBirth',
     };
 
+    var bundleProductsWithEvent = false;
+
+    // A purchase event can either log a single event with all products
+    // or multiple purchase events (one per product)
     function logPurchaseEvent(event) {
         var reportEvent = false;
-        if (event.ProductAction.ProductList) {
-            event.ProductAction.ProductList.forEach(function (product) {
-                if (product.Attributes == null) {
-                    product.Attributes = {};
+
+        if (bundleProductsWithEvent) {
+            reportEvent = logSinglePurchaseEventWithProducts(event);
+        } else {
+            reportEvent = logPurchaseEventPerProduct(event);
+        }
+        return reportEvent === true;
+    }
+
+    function logSinglePurchaseEventWithProducts(event) {
+        var quantity = 1;
+        var eventAttributes = mergeObjects(event.EventAttributes, {
+            products: [],
+        });
+        if (event.ProductAction.TransactionId) {
+            eventAttributes['Transaction Id'] =
+                event.ProductAction.TransactionId;
+        }
+
+        if (event.ProductAction.ProductList.length) {
+            event.ProductAction.ProductList.forEach(function(_product) {
+                var product = getSanitizedCustomProperties(_product);
+
+                var productName;
+                if (forwarderSettings.forwardSkuAsProductName === 'True') {
```

Comment:
Can we make `True` and `False` enums or constants? I'm fine ignoring this if you think it's going to be too far out of scope. It's just something I think we should get into the habit of doing.

Reply:
I'd say out of scope. I like the idea though that it should be an enum perhaps on the global mParticle object though, so that we don't have to keep redefining it in each kit.